aiomonitor/monitor.py

```diff
@@ -141,6 +153,72 @@ def _server(self) -> None:
                 except (socket.timeout, OSError):
                     continue
 
+    def _interactive_loop(self, sin: IO[str], sout: IO[str]) -> None:
+        """Main interactive loop of the monitor"""
+        self._sin = sin
```

Comment:
why do we need expose `self._sin` at class level?

Reply:
just to avoid `do_where(self, sin: IO[str], sout: IO[str], taskid: int)` having extra agrs? How we can ensure that _sin/_sout are set? may be add check in `_command_dispatch`

Reply:
Yes, avoiding passing the same IO arguments everywhere was the idea.

Given how the Monitor class is structured, the `do_*` methods are only called from within `_command_dispatch`, which itself is only called from within a running `_interactive_loop`. This loop sets the IO attributes at the beginning, and unsets them in the `finally` clause at the end. So they are guaranteed to be set for `do_*` methods.

If you dislike this, I'm open to reverting this.